libs/MeshKernel/src/Mesh2DIntersections.cpp

```diff
@@ -228,9 +256,17 @@ void Mesh2DIntersections::Compute(const std::vector<Point>& polyLine)
     const auto polyLineSize = static_cast<UInt>(polyLine.size());
 
     std::vector<double> cumulativeLength(polyLine.size(), 0.0);
+    std::vector<BoundingBox> polyLineBoundingBoxes(polyLine.size() - 1);
     for (UInt i = 1; i < polyLineSize; ++i)
     {
         cumulativeLength[i] = cumulativeLength[i - 1] + ComputeDistance(polyLine[i], polyLine[i - 1], m_mesh.m_projection);
+
+        const auto lowerLeftX = std::min(polyLine[i - 1].x, polyLine[i].x);
+        const auto lowerLeftY = std::min(polyLine[i - 1].y, polyLine[i].y);
+        const auto upperRightX = std::max(polyLine[i - 1].x, polyLine[i].x);
+        const auto upperRightY = std::max(polyLine[i - 1].y, polyLine[i].y);
+
+        polyLineBoundingBoxes[i - 1] = BoundingBox({lowerLeftX, lowerLeftY}, {upperRightX, upperRightY});
     }
```

Comment:
I wonder is it possible to write a static function in the BoundingBox class (or perhaps Operations) that can compute if the bounding box of two lines (given by four points) overlap.  There would be more work involved in determining if the BBs overlap, but there would be less data to move around